addon-entwicklung.md

```diff
@@ -1 +1,169 @@
+## package.yml
+
+Die Datei "package.yml" ist die einzige und erforderliche Kernkomponente eines AddOns.  
+Bei Bedarf können dort weitere Komponenten referenziert werden.
+
+<a name="anker-yaml-required"></a>
+### Erforderliche Angaben
+
+Die einzigen erforderlichen Angaben innerhalb der Datei sind Folgende:
+```
+package: test
+version: '1.0'
+```
+Die Werte müssen natürlich angepasst werden, die unbedingt erforderlichen Angaben sind somit "package" und "version".
+
+<a name="anker-yaml-optional"></a>
+### Optionale Angaben
+@TODO
+
+<a name="anker-files"></a>
+## Optionale Dateien
+Redaxo bearbeitet einige Ordner und Dateien in AddOns automatisch, darüber hinaus ist jeder Entwickler natürlich
+frei, beliebige Datein in einem AddOn bereit zu stellen.  
+Nachfolgend werden die Dateien und Ordner gelistet, nach denen innerhalb eines installierten AddOns von Redaxo automatisch gesucht wird:
+* **help.php**  
+  Ist eine Datei "help.php" im AddOn-Ordner enthalten wird diese automatisch im AddOn-Manager eingebunden und die Ausgabe angezeigt.  
+  Ist die Datei nicht enthalten wird im AddOn-Manager ein Hinweis angezeigt, daß die Datei fehlt.  
+  Es sollten Hinweise zur Funktionalität des AddOns enthalten sein, auch Versionshinweise bzw. Changelog können angezeigt werden.
+* **install.php**  
+  Ist eine Datei "install.php" im AddOn-Ordner enthalten wird diese automatisch bei der AddOn-Installation ausgeführt.  
+  Die dort integrierte Funktionalität kann viele Aktionen umfassen und ist dem Entwickler somit vollkommen freigstellt.  
+  Übliche Aktionen umfassen das Erstellen von Ordnern, das Kopieren bzw. Erstellen von Dateien und Datenbankoperationen.
+* **install.sql**  
+  Ist eine Datei "install.sql" im AddOn-Ordner enthalten wird diese automatisch bei der AddOn-Installation ausgeführt.  
+  Entwickler sollten sicherstellen, daß möglicherweise vorhandene Daten nicht überschrieben oder gelöscht werden.  
+  Übliche Aktionen sind die Erstellung von Tabellen in der Datenbank, aber es können auch Daten eingefügt werden oder alle erdenklichen SQL-Befehle integriert werden.
+* **update.php**  
+  Ist eine Datei "update.php" im AddOn-Ordner enthalten wird diese automatisch bei einem AddOn-Update ausgeführt.  
+  Auslöser für die Anzeige eines verfügbaren Updates ist die Version in der Datei "package.yml" im Vergleich zu der in einem Upload oder Download.
+* **uninstall.php**  
+  Ist eine Datei "uninstall.php" im AddOn-Ordner enthalten wird diese automatisch bei der AddOn-Deinstallation ausgeführt.  
+  Übliche Aktionen sind das Löschen von Dateien und Ordnern die zum Cachen angelegt wurden.  
+  Einige AddOns führen beim Deinstallieren auch Datenbankaktionen durch.  
+  Ein Entwickler sollte berücksichtigen, daß ein Löschen von Daten trotz Deinstallation des AddOns ungewünscht sein kann.
+* **boot.php**  
+  Ist eine Datei "boot.php" im AddOn-Ordner enthalten wird diese automatisch bei jedem Aufruf des AddOns ausgeführt, dies
```

Comment:
Wann passieren diese Aufruf? Würde ich genauer beschreiben.

Reply:
Ok, das sind ein paar ernstzunehmende Hausaufgaben ;-)
Falls Du die Antworten weißt, währe es nicht schlecht sie hier zu schreiben ;-)

Reply:
Ich bin leider bewusst nicht in der Doku-Gruppe aktiv, weil mir die Zeit fehlt, sorry 😞 
Aber gemeinsam kriegt man den Artikel bestimmt nach und nach weiter geschliffen. Muss ja nicht sofort sein.

Reply:
ok, generell wärs auch nicht schlecht manche dinge per icons anzuzeigen, z.B. Verhalten bei Reinstall, aber erst mal Fakten sammeln und zwar keine alternativen ;D